common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -0,0 +1,128 @@
+     * @return An android Intent which will be used by Android to create an AuthorizationActivity
+     */
+    public static Intent getAuthorizationActivityIntent(final Context context,
+                                                        final Intent authIntent,
+                                                        final String requestUrl,
+                                                        final String redirectUri,
+                                                        final HashMap<String, String> requestHeaders,
+                                                        final AuthorizationAgent authorizationAgent,
+                                                        final boolean webViewZoomEnabled,
+                                                        final boolean webViewZoomControlsEnabled) {
+        Intent intent;
+        final LibraryConfiguration libraryConfig = LibraryConfiguration.getInstance();
+        if (ProcessUtil.isBrokerProcess(context)) {
+            intent = new Intent(context, BrokerAuthorizationActivity.class);
+        } else {
+            if (libraryConfig.isAuthorizationInCurrentTask()) {
+                intent = new Intent(context, CurrentTaskAuthorizationActivity.class);
+            } else {
+                intent = new Intent(context, AuthorizationActivity.class);
+            }
+        }
+
+        intent.putExtra(AUTH_INTENT, authIntent);
+        intent.putExtra(REQUEST_URL, requestUrl);
+        intent.putExtra(REDIRECT_URI, redirectUri);
+        intent.putExtra(REQUEST_HEADERS, requestHeaders);
+        intent.putExtra(AUTHORIZATION_AGENT, authorizationAgent);
+        intent.putExtra(WEB_VIEW_ZOOM_CONTROLS_ENABLED, webViewZoomControlsEnabled);
+        intent.putExtra(WEB_VIEW_ZOOM_ENABLED, webViewZoomEnabled);
+        intent.putExtra(DiagnosticContext.CORRELATION_ID, DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
+        return intent;
+    }
+
+    /**
+     * Returns the correct authorization fragment for local (non-broker) authorization flows.
+     * Fragments include:
+     * {@link WebViewAuthorizationFragment}
+     * {@link BrowserAuthorizationFragment}
+     * {@link CurrentTaskBrowserAuthorizationFragment}
+     *
+     * @param intent
+     * @return returns an Fragment that's used as to authorize a token request.
+     */
+    public static Fragment getAuthorizationFragmentFromStartIntent(@NonNull final Intent intent) {
+        Fragment fragment;
+        final AuthorizationAgent authorizationAgent = (AuthorizationAgent) intent.getSerializableExtra(AUTHORIZATION_AGENT);
+        Telemetry.emit(new UiStartEvent().putUserAgent(authorizationAgent));
+
+        final LibraryConfiguration libraryConfig = LibraryConfiguration.getInstance();
+
+        if (authorizationAgent == AuthorizationAgent.WEBVIEW) {
+            fragment = new WebViewAuthorizationFragment();
```

Comment:
probably don't need to define a variable for `fragment`...can just return the instance directly from each of the branches here

Reply:
Sure.  I can fix this in the existing code to... ;-)